listenbrainz/webserver/static/js/src/user-feed/UserFeed.tsx

```diff
@@ -327,9 +329,94 @@ export default class UserFeedPage extends React.Component<
     this.setState({ recordingFeedbackMap: newFeedbackMap });
   };
 
+  deleteFeedEvent = async (event: TimelineEvent) => {
+    const { currentUser, APIService } = this.context;
+    const { newAlert } = this.props;
+    const { events } = this.state;
+    if (
+      event.event_type === EventType.RECORDING_RECOMMENDATION ||
+      event.event_type === EventType.NOTIFICATION
+    ) {
+      try {
+        const status = await APIService.deleteFeedEvent(
+          event.event_type,
+          currentUser.name,
+          currentUser.auth_token as string,
+          event.id
+        );
+        if (status === 200) {
+          newAlert(
+            "success",
+            "Successfully deleted!",
+            <>Successfully deleted!</>
+          );
+          const new_events = _remove(events, (element) => {
+            // Making sure the event that is getting deleted is either a recommendation or notification
+            // Since, recommendation and notification are in same db, and might have same id as a pin
+            // Similarly we later on filter by id and event_type for pin deletion
+            return !(
+              element?.id === event.id &&
+              (element.event_type === EventType.RECORDING_RECOMMENDATION ||
+                element.event_type === EventType.NOTIFICATION)
+            );
+          });
+          this.setState({ events: new_events });
+        }
+      } catch (error) {
+        newAlert(
+          "danger",
+          "Could not delete event",
+          <>
+            Something went wrong when we tried to delete your event, please try
+            again or contact us if the problem persists.
+            <br />
+            <strong>
+              {error.name}: {error.message}
+            </strong>
+          </>
+        );
+      }
+    } else if (event.event_type === EventType.RECORDING_PIN) {
+      try {
+        const status = await APIService.deletePin(
+          currentUser.auth_token as string,
+          event.id as number
+        );
+        if (status === 200) {
+          newAlert(
+            "success",
+            "Successfully deleted!",
+            <>Successfully deleted!</>
+          );
+          const new_events = _remove(events, (element) => {
+            return !(
+              element?.id === event.id &&
+              element.event_type === EventType.RECORDING_PIN
+            );
+          });
+          this.setState({ events: new_events });
```

Comment:
At a glance this isn't the utility you're looking for.
According to the docs (https://lodash.com/docs/4.17.15#remove), Lodash's remove mutates the array (which we don't want) and returns an array of the removed values (which isn't exactly what we need either.
I think that's why you had to invert the value (`return !(…`), which produces the results we want but is difficult to read.

Consider using _.filter instead (https://lodash.com/docs/4.17.15#filter) which returns "an array of all elements predicate returns truthy for" AND doesn't mutate the original array.

Reply:
seems like the boolean nature of both `remove` and `filter` are same
![image](https://user-images.githubusercontent.com/11733600/152238346-ce13551a-413b-432c-8554-d50b90bf4fae.png)
what we can do for better code readability is this, and not mess around with `new_events` at all (this mutates the original array)
```js
         _remove(events, (element) => {
            return !(
              element?.id === event.id &&
              element.event_type === EventType.RECORDING_PIN
            );
          });
          this.setState({ events: events });
```


I also wanted to ask, why is mutating the array a bad idea?